blocks/library/list/index.js

```diff
@@ -108,31 +50,122 @@ registerBlockType( 'core/list', {
 		],
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { nodeName = 'OL', values = [] } = attributes;
-		return (
-			<Editable
-				tagName={ nodeName.toLowerCase() }
-				getSettings={ ( settings ) => ( {
-					...settings,
-					plugins: ( settings.plugins || [] ).concat( 'lists' ),
-					lists_indent_on_tab: false,
-				} ) }
-				onSetup={ ( editor ) => {
-					editor.on( 'nodeChange', ( nodeInfo ) => {
-						setAttributes( { internalListType: findInternalListType( nodeInfo ) } );
-					} );
-					setAttributes( { editor } );
-				} }
-				onChange={ ( nextValues ) => {
-					setAttributes( { values: nextValues } );
-				} }
-				value={ values }
-				focus={ focus }
-				onFocus={ setFocus }
-				showAlignments
-				className="blocks-list" />
-		);
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.setupEditor = this.setupEditor.bind( this );
+			this.getEditorSettings = this.getEditorSettings.bind( this );
+			this.setNextValues = this.setNextValues.bind( this );
+
+			this.state = {
+				internalListType: null,
+			};
+		}
+
+		isListActive( listType ) {
+			const { internalListType } = this.state;
+			const { nodeName = 'OL' } = this.props.attributes;
+
+			return listType === ( internalListType ? internalListType : nodeName );
+		}
+
+		findInternalListType( { parents } ) {
+			const list = find( parents, ( node ) => node.nodeName === 'UL' || node.nodeName === 'OL' );
+			return list ? list.nodeName : null;
+		}
+
+		setupEditor( editor ) {
+			editor.on( 'nodeChange', ( nodeInfo ) => {
+				this.setState( {
+					internalListType: this.findInternalListType( nodeInfo ),
+				} );
+			} );
+
+			this.editor = editor;
+		}
+
+		createSetListType( type, command ) {
+			return () => {
+				const { setAttributes } = this.props;
+				const { internalListType } = this.state;
+				if ( internalListType ) {
+					// only change list types, don't toggle off internal lists
+					if ( internalListType !== type && this.editor ) {
+						this.editor.execCommand( command );
+					}
+				} else {
+					setAttributes( { nodeName: type } );
+				}
+			};
+		}
+
+		createExecCommand( command ) {
+			return () => {
+				if ( this.editor ) {
+					this.editor.execCommand( command );
+				}
+			};
+		}
+
+		getEditorSettings( settings ) {
+			return {
+				...settings,
+				plugins: ( settings.plugins || [] ).concat( 'lists' ),
+				lists_indent_on_tab: false,
+			};
+		}
+
+		setNextValues( nextValues ) {
+			this.props.setAttributes( { values: nextValues } );
+		}
+
+		render() {
+			const { attributes, focus, setFocus } = this.props;
+			const { nodeName = 'OL', values = [] } = attributes;
+
+			return [
+				<BlockControls
```

Comment:
>I think you missed the `focus` check here ;)

How embarrassing considering I'd just remarked on how I'd been okay with it in the previous pull request 😄